pkg/detectors/azuresearchadminkey/azuresearchadminkey.go

```diff
@@ -0,0 +1,100 @@
+package azuresearchadminkey
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat     = regexp.MustCompile(detectors.PrefixRegex([]string{"azure"}) + `\b([0-9a-zA-Z]{52})\b`)
+	servicePat = regexp.MustCompile(detectors.PrefixRegex([]string{"azure"}) + `\b([0-9a-zA-Z]{7,40})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"azure"}
+}
+
+// FromData will find and optionally verify AzureSearchAdminKey secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	serviceMatches := servicePat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+		for _, serviceMatch := range serviceMatches {
+			if len(serviceMatch) != 2 {
+				continue
+			}
+			resServiceMatch := strings.TrimSpace(serviceMatch[1])
+
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_AzureSearchAdminKey,
+				Raw:          []byte(resMatch + resServiceMatch),
```

Comment:
Small correction to what I told you previously: Raw should contain the key match, and RawV2 should contain all components. This is because we use part of the Raw field to provide a redacted value to help people find the leak.
In this case it starts with the key value, so it will be fine as-is.